src/Features/Core/Portable/ExtractMethod/ExtractMethodResult.cs

```diff
@@ -27,69 +22,32 @@ internal abstract class ExtractMethodResult
         /// </summary>
         public ImmutableArray<string> Reasons { get; }
 
-        /// <summary>
-        /// The transformed document that was produced as a result of the extract method operation.
-        /// </summary>
-        public Document? DocumentWithoutFinalFormatting { get; }
-
-        /// <summary>
-        /// Formatting rules to apply to <see cref="DocumentWithoutFinalFormatting"/> to obtain the final formatted
-        /// document.
-        /// </summary>
-        public ImmutableArray<AbstractFormattingRule> FormattingRules { get; }
-
-        /// <summary>
-        /// The name token for the invocation node that replaces the extracted code.
-        /// </summary>
-        public SyntaxToken? InvocationNameToken { get; }
+        private readonly AsyncLazy<(Document document, SyntaxToken? invocationNameToken)>? _lazyData;
```

Comment:
computation fo the final doc, and token to rename, is done lazily and cached here.